addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/PackageContentAccessHandler.java

```diff
@@ -0,0 +1,425 @@
+
+public class PackageContentAccessHandler
+                extends ContentAccessHandler
+{
+    public static final String CROSS = "-";
+
+    public static final String PATH_SPLITER = "/";
+
+    public static final String TGZ = ".tgz";
+
+    public static final String PACAGE_TGZ = "package" + TGZ;
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata, final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
+                       builderModifier );
+    }
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
+                       null );
+    }
+
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final String version, final String tarball, final Boolean cacheOnly, final String baseUri,
+                            final HttpServletRequest request, EventMetadata eventMetadata )
+    {
+        return doHead( packageType, type, name, packageName, version, tarball, cacheOnly, baseUri, request,
+                       eventMetadata, null );
+    }
+
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final String version, final String tarball, final Boolean cacheOnly, final String baseUri,
+                            final HttpServletRequest request, EventMetadata eventMetadata,
+                            final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( !PackageTypes.contains( packageType ) )
+        {
+            Response.ResponseBuilder builder = Response.status( 400 );
+            if ( builderModifier != null )
+            {
+                builderModifier.accept( builder );
+            }
+            return builder.build();
+        }
+
+        final StoreType st = StoreType.get( type );
+        final StoreKey sk = new StoreKey( packageType, st, name );
+
+        eventMetadata = eventMetadata.set( ContentManager.ENTRY_POINT_STORE, sk );
+
+        final AcceptInfo acceptInfo = jaxRsRequestHelper.findAccept( request, ApplicationContent.text_html );
+
+        Response response = null;
+
+        if ( packageName == null || packageName.trim().length() < 1 )
```

Comment:
@jdcasey Yeah, I know what you mean here, I also wanna try to make the npm content hanldler in ContentAccessHandler at the beginning, And the npm doHead method is exatly adding some path mapping stuff compared with the ContentAccessHandler one, while doGet method has many difference with the ContentAccessHandler one, such as it will not handle some situation like direcotry/index.html render, but need to do some string(package.json output) or inputstream(.tgz output) response. So if we wanna merge all the contentHandler code here to reduce duplicated code (I also deem it has to much duplication in doHead), I could try to make the path mapping  handler in REST API methods, and for doGet, I'll and a specific npm packageType contet response logic block, make these part more clear-cut. Thanks, sounds ok?